Sources/BarcodeView/Models/EAN13.swift

```diff
@@ -70,16 +70,18 @@ extension EAN13 {
 
         return digits
     }
-
+    
     private static func verifyChecksumDigit(value: String) -> Bool {
-        let digits = value.map { Int(String($0))! }.dropLast()
-
-        let odd = stride(from: 1, to: digits.count, by: 2).map { digits[$0] }.reduce(0, +) * 3
-        let even = stride(from: 0, to: digits.count, by: 2).map { digits[$0] }.reduce(0, +)
-
-        let calculatedChecksum = 10 - ((odd + even) % 10)
-
-        return String(calculatedChecksum) == value.last.map({ String($0) })
+        let digits = value.compactMap { Int(String($0)) }
+        let lastDigit = digits.last!
```

Comment:
This seems inherently unsafe if after the `compactMap` there are zero elements in `digits`. Previous code was written a long time ago I should have probably made it safer long ago 😂 Haven't looked at this code in awhile so maybe there are other verification steps that make sure it's the proper number of digits but you may want to check that `value.count == digits.count`

Reply:
@nrivard yes, I've seen it, but on line 16 you checked digits, so it is actually fine:

https://github.com/nrivard/BarcodeView/blob/1c18b3236db7ac03146981d26a8258b5e62117c8/Sources/BarcodeView/Models/EAN13.swift#L16